py-polars/tests/unit/operations/test_group_by.py

```diff
@@ -768,6 +768,13 @@ def test_group_by_partitioned_ending_cast(monkeypatch: Any) -> None:
     assert_frame_equal(out, expected)
 
 
+def test_group_by_series_partitioned() -> None:
+    # test 15354
+    df = pl.DataFrame([0] * 1000)
```

Comment:
Instead of `1000` we should use `PARTITION_LIMIT` here which is available under `py-polars/polars/testing/_constants.py`

On `debug` builds the threshold is not 1000.